Support TS isolatedModules

When I try to `yarn build` my Nextjs project using typescript and vega-lite, I get this error:
```
The following mandatory changes were made to your tsconfig.json:

	- isolatedModules was set to true (requirement for babel)

Failed to compile.

./node_modules/vega-lite/src/compositemark/index.ts:28:9
Type error: Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.

  26 | } from './errorbar';
  27 |
> 28 | export {BoxPlotConfig} from './boxplot';
     |         ^
  29 | export {ErrorBandConfigMixins} from './errorband';
  30 | export {ErrorBarConfigMixins} from './errorbar';
  31 |
error Command failed with exit code 1.
```
I think this is pointing to this line: https://github.com/vega/vega-lite/blob/22057b18d54e71be4cc7c959c5f83d43d765a697/src/compositemark/index.ts#L28
Do we need to update these `export` statements to `export type`?

I doubt that the vega-lite codebase is not compatible with Babel, since vega-lite is currently producing ES2015 output. So maybe my TypeScript configuration is wrong?

Here is my `tsconfig.json`:
```
{
  "compilerOptions": {
    "target": "es6",
    "lib": [
      "dom",
      "dom.iterable",
      "esnext"
    ],
    "allowJs": true,
    "skipLibCheck": true,
    "strict": false,
    "forceConsistentCasingInFileNames": true,
    "noEmit": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "jsx": "preserve",
    "baseUrl": ".",
    "isolatedModules": true,
    "downlevelIteration": true
  },
  "include": [
    "next-env.d.ts",
    "**/*.ts",
    "**/*.tsx"
  ],
  "exclude": [
    "node_modules"
  ]
}
```

Thanks so much for taking a look! Sure, here's a simpler reproduction of the issue:

src/test.ts:
```ts
import { TopLevelSpec } from 'vega-lite'
import { Transform } from 'vega-lite/src/transform'
import { StringFieldDef } from 'vega-lite/src/channeldef'

function main(): TopLevelSpec {
  return {
    encoding: {
      x: {
        field: 'year',
        type: 'temporal',
      },
      y: {
        field: 'value',
        type: 'quantitative',
      },
    },
    layer: [
      {
        mark: 'bar',
        encoding: {
          x: {
            field: 'year'
          },
          y: {
            field: 'value'
          },
        },
      }
    ],
  }
}
```
package.json:
```json
{
  "name": "typescript-test",
  "version": "1.0.0",
  "main": "index.js",
  "license": "MIT",
  "devDependencies": {
    "typescript": "^4.5.4"
  },
  "dependencies": {
    "vega": "^5.21.0",
    "vega-lite": "^5.2.0"
  }
}
```
This works with the following `tsconfig.json`:
```json
{
  "compilerOptions": {
    "target": "es6",
    "lib": ["es2020", "dom"],
    "module": "commonjs",
    // "isolatedModules": true,
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true,
    "strict": false,
    "skipLibCheck": true
  }
}
```
but gives this error when I uncomment `// "isolatedModules": true,`:
```
$ /Users/sidharth.kapur/personal-workspace/typescript-test/node_modules/.bin/tsc -p tsconfig.json
node_modules/vega-lite/src/compositemark/index.ts:28:9 - error TS1205: Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.

28 export {BoxPlotConfig} from './boxplot';
           ~~~~~~~~~~~~~

node_modules/vega-lite/src/compositemark/index.ts:29:9 - error TS1205: Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.

29 export {ErrorBandConfigMixins} from './errorband';
           ~~~~~~~~~~~~~~~~~~~~~

node_modules/vega-lite/src/compositemark/index.ts:30:9 - error TS1205: Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.

30 export {ErrorBarConfigMixins} from './errorbar';
           ~~~~~~~~~~~~~~~~~~~~

node_modules/vega-lite/src/spec/base.ts:15:9 - error TS1205: Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.

15 export {TopLevel} from './toplevel';
           ~~~~~~~~

node_modules/vega-lite/src/spec/index.ts:17:9 - error TS1205: Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.

17 export {BaseSpec, LayoutSizeMixins} from './base';
           ~~~~~~~~

node_modules/vega-lite/src/spec/index.ts:17:19 - error TS1205: Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.

17 export {BaseSpec, LayoutSizeMixins} from './base';
                     ~~~~~~~~~~~~~~~~

node_modules/vega-lite/src/spec/index.ts:19:3 - error TS1205: Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.

19   GenericHConcatSpec,
     ~~~~~~~~~~~~~~~~~~

node_modules/vega-lite/src/spec/index.ts:20:3 - error TS1205: Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.

20   GenericVConcatSpec,
     ~~~~~~~~~~~~~~~~~~

node_modules/vega-lite/src/spec/index.ts:24:3 - error TS1205: Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.

24   NormalizedConcatSpec
     ~~~~~~~~~~~~~~~~~~~~

node_modules/vega-lite/src/spec/index.ts:26:9 - error TS1205: Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.

26 export {GenericFacetSpec, isFacetSpec, NormalizedFacetSpec} from './facet';
           ~~~~~~~~~~~~~~~~

node_modules/vega-lite/src/spec/index.ts:26:40 - error TS1205: Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.

26 export {GenericFacetSpec, isFacetSpec, NormalizedFacetSpec} from './facet';
                                          ~~~~~~~~~~~~~~~~~~~

node_modules/vega-lite/src/spec/index.ts:27:9 - error TS1205: Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.

27 export {GenericLayerSpec, isLayerSpec, LayerSpec, NormalizedLayerSpec} from './layer';
           ~~~~~~~~~~~~~~~~

node_modules/vega-lite/src/spec/index.ts:27:40 - error TS1205: Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.

27 export {GenericLayerSpec, isLayerSpec, LayerSpec, NormalizedLayerSpec} from './layer';
                                          ~~~~~~~~~

node_modules/vega-lite/src/spec/index.ts:27:51 - error TS1205: Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.

27 export {GenericLayerSpec, isLayerSpec, LayerSpec, NormalizedLayerSpec} from './layer';
                                                     ~~~~~~~~~~~~~~~~~~~

node_modules/vega-lite/src/spec/index.ts:28:23 - error TS1205: Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.

28 export {isRepeatSpec, RepeatSpec} from './repeat';
                         ~~~~~~~~~~

node_modules/vega-lite/src/spec/index.ts:29:9 - error TS1205: Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.

29 export {TopLevel} from './toplevel';
           ~~~~~~~~

node_modules/vega-lite/src/spec/index.ts:30:9 - error TS1205: Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.

30 export {FacetedUnitSpec, GenericUnitSpec, isUnitSpec, NormalizedUnitSpec, UnitSpec} from './unit';
           ~~~~~~~~~~~~~~~

node_modules/vega-lite/src/spec/index.ts:30:26 - error TS1205: Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.

30 export {FacetedUnitSpec, GenericUnitSpec, isUnitSpec, NormalizedUnitSpec, UnitSpec} from './unit';
                            ~~~~~~~~~~~~~~~

node_modules/vega-lite/src/spec/index.ts:30:55 - error TS1205: Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.

30 export {FacetedUnitSpec, GenericUnitSpec, isUnitSpec, NormalizedUnitSpec, UnitSpec} from './unit';
                                                         ~~~~~~~~~~~~~~~~~~

node_modules/vega-lite/src/spec/index.ts:30:75 - error TS1205: Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.

30 export {FacetedUnitSpec, GenericUnitSpec, isUnitSpec, NormalizedUnitSpec, UnitSpec} from './unit';
                                                                             ~~~~~~~~


Found 20 errors.

error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

The errors also go away if I remove the 'vega-lite/src/transform' and 'vega-lite/src/channeldef' imports.
Maybe I'm importing these modules incorrectly?

--- a/src/compositemark/index.ts
+++ b/src/compositemark/index.ts
@@ -25,9 +25,9 @@ import {
   normalizeErrorBar
 } from './errorbar';
 
-export {BoxPlotConfig} from './boxplot';
-export {ErrorBandConfigMixins} from './errorband';
-export {ErrorBarConfigMixins} from './errorbar';
+export type {BoxPlotConfig} from './boxplot';
+export type {ErrorBandConfigMixins} from './errorband';
+export type {ErrorBarConfigMixins} from './errorbar';
 
 export type CompositeMarkNormalizerRun = (
   spec: GenericUnitSpec<any, any>,

--- a/src/spec/base.ts
+++ b/src/spec/base.ts
@@ -12,7 +12,7 @@ import {LayoutAlign, RowCol} from '../vega.schema';
 import {isConcatSpec, isVConcatSpec} from './concat';
 import {isFacetMapping, isFacetSpec} from './facet';
 
-export {TopLevel} from './toplevel';
+export type {TopLevel} from './toplevel';
 
 /**
  * Common properties for all types of specification

--- a/src/spec/index.ts
+++ b/src/spec/index.ts
@@ -14,20 +14,26 @@ import {RepeatSpec} from './repeat';
 import {TopLevel} from './toplevel';
 import {FacetedUnitSpec, GenericUnitSpec, NormalizedUnitSpec, TopLevelUnitSpec, UnitSpecWithFrame} from './unit';
 
-export {BaseSpec, LayoutSizeMixins} from './base';
-export {
+export type {BaseSpec, LayoutSizeMixins} from './base';
+export type {
   GenericHConcatSpec,
   GenericVConcatSpec,
+  NormalizedConcatSpec
+} from './concat';
+export {
   isAnyConcatSpec,
   isHConcatSpec,
   isVConcatSpec,
-  NormalizedConcatSpec
 } from './concat';
-export {GenericFacetSpec, isFacetSpec, NormalizedFacetSpec} from './facet';
-export {GenericLayerSpec, isLayerSpec, LayerSpec, NormalizedLayerSpec} from './layer';
-export {isRepeatSpec, RepeatSpec} from './repeat';
-export {TopLevel} from './toplevel';
-export {FacetedUnitSpec, GenericUnitSpec, isUnitSpec, NormalizedUnitSpec, UnitSpec} from './unit';
+export type {GenericFacetSpec, NormalizedFacetSpec} from './facet';
+export {isFacetSpec} from './facet';
+export type {GenericLayerSpec, LayerSpec, NormalizedLayerSpec} from './layer';
+export {isLayerSpec} from './layer';
+export type {RepeatSpec} from './repeat';
+export {isRepeatSpec} from './repeat';
+export type {TopLevel} from './toplevel';
+export type {FacetedUnitSpec, GenericUnitSpec, NormalizedUnitSpec, UnitSpec} from './unit';
+export {isUnitSpec} from './unit';
 
 /**
  * Any specification in Vega-Lite.

--- a/test/jest.overrides.ts
+++ b/test/jest.overrides.ts
@@ -11,3 +11,5 @@ console.warn = function (msg: string, msg2: string) {
   warn.apply(console, arguments);
   throw new Error(`${msg}: ${msg2} -- Please remove unnecessary errors or use log.wrap to consume reasonable errors`);
 };
+
+export {};

--- a/tsconfig.build.json
+++ b/tsconfig.build.json
@@ -5,7 +5,8 @@
     "noEmit": false,
     "declaration": true,
     "declarationMap": true,
-    "sourceMap": true
+    "sourceMap": true,
+    "isolatedModules": true
   },
   "include": []
 }

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -14,6 +14,7 @@
     "preserveConstEnums": true,
     "resolveJsonModule": true,
     "suppressImplicitAnyIndexErrors": true,
+    "isolatedModules": true,
     "lib": ["ESNext.Array", "DOM", "DOM.Iterable"],
     "noEmit": true
   },

